docs/guide/declarative_containers.md

```diff
@@ -0,0 +1,111 @@
+# Declarative Environment Configuration
+
+jobq allows you to define your Docker environment in a declarative YAML format. Instead of writing a Dockerfile directly.
+
```

Comment:
"Docker environment" leads me in a different direction that what we're trying to achieve: define the contents and requirements for a Docker image build (the environment for me would be the Docker daemon and other stuff that happens on the host machine)

Reply:
Tbh, I struggled writing that because someone who has working associations to all that would likely not use our `.yaml` format but rather supply a Dockerfile directly. So I was trying to aim the explanations at someone who has but a basic familiarity with containerization. 

I do agree though that overloading of vocabulary is bad as we make thorough understanding harder for newbies down the road.

I would, however, avoid diving into the daemon etc. within the docs.